match/spec/storage/s3_storage_spec.rb

```diff
@@ -1,111 +1,144 @@
 describe Match do
   describe Match::Storage::S3Storage do
-    subject { described_class.new(s3_region: nil, s3_access_key: nil, s3_secret_access_key: nil, s3_bucket: 'foobar') }
-    let(:working_directory) { '/var/folders/px/abcdefghijklmnop/T/d20181026-96528-1av4gge' }
 
-    before do
-      allow(subject).to receive(:working_directory).and_return(working_directory)
-      allow(subject).to receive(:s3_client).and_return(s3_client)
-    end
+    describe "official endpoint" do
```

Comment:
I did some research. For different subjects, we can only use an inner scope of describe to make sure there's no conflict.

https://stackoverflow.com/a/11744451

Reply:
Makes sense!

Since the "official endpoint" is the most standard/common path, do you mind reverting this change and then putting your new code in an equivalent `describe "custom endpoint" do`? 🙏

Reply:
line 116 is already the `describe "custom endpoint" do`

Without wrapping the "official endpoint" into a "describe scope" it will fail.
This is my local testing output after reverting it:

```
 #initialize with custom endpoint and force_path_style
      passes endpoint and force_path_style to S3ClientHelper (FAILED - 1)

Failures:

  1) Match Match::Storage::S3Storage#initialize with custom endpoint and force_path_style passes endpoint and force_path_style to S3ClientHelper
     Failure/Error: allow(subject).to receive(:s3_client).and_return(s3_client)
     
     NameError:
       undefined local variable or method `s3_client' for #<RSpec::ExampleGroups::Match::MatchStorageS3Storage::InitializeWithCustomEndpointAndForcePathStyle "passes endpoint and force_path_style to S3ClientHelper" (./match/spec/storage/s3_storage_spec.rb:129)>
     # ./match/spec/storage/s3_storage_spec.rb:8:in `block (3 levels) in <top (required)>'

Finished in 0.01318 seconds (files took 1.49 seconds to load)
9 examples, 1 failure

Failed examples:

rspec ./match/spec/storage/s3_storage_spec.rb:129 # Match Match::Storage::S3Storage#initialize with custom endpoint and force_path_style passes endpoint and force_path_style to S3ClientHelper
```

Reply:
gentle ping @rogerluan 

thanks!

Reply:
gentle ping @rogerluan